### PR TITLE
Horizontal Resolution

### PR DESCRIPTION
**Pull request by machin3io**
* add width argument, to set the horizintal resolution
* add example command using width + frame input

### CODE REVIEW

File: gifgen [hunk 1]
```diff
@@ -2,32 +2,35 @@
...
   echo "  $ gifgen -o demo.gif SCM_1457.mp4"
   echo "  $ gifgen -sf 15 screencap.mov"
+  echo "  $ gifgen -w 700 -s -o my.gif frames/frame_%4d.png"
 }
 
 # Setup defaults
 pid=$$
 palette="/tmp/gif-palette-$pid.png"
 fps="10"
+width="640"
```
Comment (lukechilds) [inline gifgen:27]: We shouldn't default to a custom width. It should use the input width unless specified otherwise.

File: gifgen [hunk 2]
```diff
@@ -2,32 +2,35 @@
 
 # Echo help/usage message
 show_help() {
-  echo "gifgen 1.1.2"
+  echo "gifgen 1.2.0"
```
Comment (lukechilds) [inline gifgen:5]: No need to update version numbers in this PR. I'll do a release separately.

File: gifgen [hunk 3]
```diff
@@ -2,32 +2,35 @@
...
   echo "  -f   Frames per second [10]"
+  echo "  -w   Width, horizontal resolution [640]"
   echo "  -s   Optimize for static background"
   echo "  -v   Display verbose output from ffmpeg"
   echo
   echo "Examples:"
   echo "  $ gifgen video.mp4"
   echo "  $ gifgen -o demo.gif SCM_1457.mp4"
   echo "  $ gifgen -sf 15 screencap.mov"
+  echo "  $ gifgen -w 700 -s -o my.gif frames/frame_%4d.png"
```
Comment (lukechilds) [inline gifgen]: Can you change this to a simpler example? It should just demo the -w option. Something like:

```
$ gifgen -w 700 -o scaled.gif screencap.mov
```

File: README.md [hunk 4]
```diff
@@ -25,21 +25,23 @@ Usage: gifgen [options] [input]
...
+  $ gifgen -w 700 -s -o my.gif frames/frame_%4d.png
 ```
 
 ## Installation
 
 ### macOS
 
 ```shell
-brew install lukechilds/tap/gifgen
+brew install machin3io/tap/gifgen
```
Comment (lukechilds) [inline README.md]: Did you mean to add this?

**Review by lukechilds — CHANGES_REQUESTED** [review]
Thanks for taking the time to put this together. Some feedback for you.

File: gifgen (hunk 1 above)
Comment (machin3io) [reply gifgen]: That's how it currently behaves. Like the 10 fps, the 640 is used when no width is passed in based on my tests.

File: README.md [hunk 5]
```diff
@@ -18,39 +18,30 @@ If you've ever tried encoding GIFs with `ffmpeg` there's a good chance your resu
...
 Examples:
   $ gifgen video.mp4
   $ gifgen -o demo.gif SCM_1457.mp4
   $ gifgen -sf 15 screencap.mov
+  $ gifgen -w 700 -o my.gif frames/frame_%4d.png
 ```
 
 ## Installation
 
-### macOS
```
Comment (lukechilds) [inline README.md:39]: Can you revert removing the installation instructions?

**Review by lukechilds — CHANGES_REQUESTED** [review]

**Comment by lukechilds** [comment]
Closed in favour of https://github.com/lukechilds/gifgen/pull/8

**Pull request closed without merging**